### PR TITLE
Gender and PVP filtering / L15 cells

### PR DESCRIPTION
**Pull request by bradders2000**
Filter pokemon by gender
Filter pokemon by great/ultra league rank (1 - 100)
Show L15 S2 cells (nearby / cell pokemon)
Enhance weather icons to show when it's out of date (requires a tweak to RDM's handling of when it updates the weather in the db)
Tweaks to menu layout
Tweaks to egg/raid/invasion timers
Specific discord users can be given a certain access level
replace iconSizeModifier with iconSize and use real values

### CODE REVIEW

**Comment by nileplumb** [comment]
DO IT DO IT

**Comment by albertehc** [comment]
Quest filter is broken

**Comment by bradders2000** [comment]
> Quest filter is broken or did i miss some configuration?

It wasn't actually the filter, but the issue has been fixed now. Thank you for testing.

**Comment by samonthenetuk** [comment]
Just trying out this branch and it is working really well. Although by default 'Pokemon Markericon Size' is blank. Is there a new setting in config to default this?

**Comment by samonthenetuk** [comment]
thanks. this is working perfectly for me now

File: login.php [hunk 1]
```diff
@@ -424,6 +430,7 @@
                     'user' => $user['name'],
                     'access_level' => $facebookAccessLevel,
                     'avatar' => $user['picture']['url'],
+                    'access_level' => null,
```
Comment (whitewillem) [inline login.php:433]: Pls undo this line

File: login.php [hunk 2]
```diff
@@ -470,6 +477,7 @@
                     'user' => $user->name,
                     'access_level' => $groupmeAccessLevel,
                     'avatar' => $user->image_url,
+                    'access_level' => null,
```
Comment (whitewillem) [inline login.php:480]: Pls undo this line

File: static/js/map.common.js [hunk 3]
```diff
@@ -492,6 +527,7 @@ var mapData = {
     pokemons: {},
     gyms: {},
     pokestops: {},
+    lurePokemons: {},
```
Comment (whitewillem) [inline static/js/map.common.js:530]: Pls undo this line

File: static/js/map.js [hunk 4]
```diff
@@ -92,9 +96,12 @@ var rawDataIsLoading = false
 var searchDelay
 var locationMarker
 var rangeMarkers = ['pokemon', 'pokestop', 'gym']
+var placementRangeMarkers = ['pokestop', 'gym']
 var storeZoom = true
 var moves
 var pokedex
+var weather
```
Comment (whitewillem) [inline static/js/map.js:103]: Pls undo this line

File: static/js/map.js [hunk 5]
```diff
@@ -92,9 +96,12 @@ var rawDataIsLoading = false
...
+var boostedMons
```
Comment (whitewillem) [inline static/js/map.js:104]: Pls undo this line

File: lib/RocketMap_MAD.php [hunk 6]
```diff
@@ -100,14 +100,32 @@ public function get_active($eids, $minIv, $minLevel, $exMinIv, $bigKarp, $tinyRa
                 $conds[] = '(cp_multiplier >= ' . $this->cpMultiplier[$minLevel] . ' OR pokemon_id IN(' . $exMinIv . ') )';
             }
         }
+        if (!empty($despawnTimeType)) {
+// expire_timestamp_verified not available?
```
Comment (whitewillem) [inline lib/RocketMap_MAD.php:104]: Mad has a join on trs_spawn. Check if ts.calc_endminsec is null or not

Comment (whitewillem) [reply lib/RocketMap_MAD.php:104]: Also... ```seen_type enum('wild', 'encounter', 'nearby_stop', 'nearby_cell', 'lure_wild', 'lure_encounter'),```

**Comment by jepke** [comment]
Upon initial map load all filter tabs are empty until i switch between them.

auto change for iconsize to new value would be nice but this is a global PMSF thing that needs some brains. Quick one would be to change the `$iconSize` var as a whole so it becomes deprecated.

Console logging is nice but i rather see something to have that disabled by default and only turned on for debugging. This is really annoying for developing in the future.

Other than this looks good. Great work

**Comment by bradders2000** [comment]
The console logging is only there to help debug the new uicons stuff. Once everyone is happy, it can be removed. Also happy to tie it in with a new uiconsDebugMode type variable. 

Does any of the logging show and clues about why your filters are empty? Are all the json files loading correctly at the start?

**Comment by jepke** [comment]
> The console logging is only there to help debug the new uicons stuff. Once everyone is happy, it can be removed. Also happy to tie it in with a new uiconsDebugMode type variable.
> 
> Does any of the logging show and clues about why your filters are empty? Are all the json files loading correctly at the start?
> 
> Changing iconSize to another variable name has come up in conversation as a way to get around people having to reset their chosen Pokemon icon size. I will sort that.

Just leave it in for future troubleshooting. Create a new PHP var `$enableJSDebug = false` in default.php. If true console log is enabled. There are some parts in JS i'd like to get console debugging from. It simplifies debugging for mappers.

**Comment by bradders2000** [comment]
Sounds like a plan. :)

**Comment by jepke** [comment]
> Sounds like a plan. :)

If you can add that and change the `iconSize` to something else i'd be happy to merge into develop unless you want to add more things.

**Comment by bradders2000** [comment]
> > Sounds like a plan. :)
> 
> If you can add that and change the `iconSize` to something else i'd be happy to merge into develop unless you want to add more things.

Ok, done.

**Pull request merged**